rust/src/dcerpc/dcerpc.rs

```diff
@@ -1113,10 +1113,10 @@ unsafe extern "C" fn get_tx_cnt(vtx: *mut std::os::raw::c_void) -> u64 {
 pub(super) unsafe extern "C" fn get_alstate_progress(tx: *mut std::os::raw::c_void, direction: u8
                                                  )-> std::os::raw::c_int {
     let tx = cast_pointer!(tx, DCERPCTransaction);
-    if direction == Direction::ToServer.into() && tx.req_done {
+    if Direction::from(direction) == Direction::ToServer && tx.req_done {
         SCLogDebug!("tx {} TOSERVER progress 1 => {:?}", tx.call_id, tx);
         return 1;
-    } else if direction == Direction::ToClient.into() && tx.resp_done {
+    } else if Direction::from(direction) == Direction::ToClient && tx.resp_done {
```

Comment:
I'm not 100% sure about this one..  `Direction::from` will default to ToServer if the direction integer value is not an exact match.

The `Direction::ToServer.into()` is just a wrapper around `as u8` which might be better to use directly, or even a member function `as_u8()`.

Reply:
I think this `get_alstate_progress` could turn the `u8` into the rust enum `Direction` and panic if we see an unknown value...

But `u8::from(Direction::toServer)` is a simpler faster change :-)

Reply:
Ok, using that over here: https://github.com/OISF/suricata/pull/12829